base/deprecated.jl

```diff
@@ -1298,6 +1298,10 @@ end
      end
 end
 
+# #19635
+@deprecate zeros{T <: Number}(::Type{T}, arr::Range)  zeros(T, length(arr))
```

Comment:
It's not just `Range`. 0.5:
```julia
julia> zeros(Int, [1,2,3,4])
4-element Array{Int64,1}:
 0
 0
 0
 0
```
master:
```julia
julia> zeros(Int, [1,2,3,4])
4-element Array{Int64,1}:
 0
 0
 0
 0
```
On 0.5, `zeros(T, arg)` ends up calling `fill!(convert(Array{T}, arg), zero(T))`. So anything that can be converted to an `Array` is a plausible input.